src/ducks/wallet/lyra.ts

```diff
@@ -0,0 +1,76 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { getPublicKey } from "@stellar/lyra-api";
+
+import {
+  ActionStatus,
+  RejectMessage,
+  WalletInitialState,
+} from "constants/types.d";
+
+export const fetchLyraStellarAddressAction = createAsyncThunk<
+  { publicKey: string },
+  undefined,
+  { rejectValue: RejectMessage }
+>(
+  "walletLyra/fetchLyraStellarAddressAction",
+  async (_, { rejectWithValue }) => {
+    try {
+      const lyraResponse = await getPublicKey();
+
+      if (lyraResponse.error) {
+        return rejectWithValue({
+          errorString: lyraResponse.error,
+        });
+      }
+
+      return { publicKey: lyraResponse.publicKey };
+    } catch (error) {
+      return rejectWithValue({
+        errorString: error.toString(),
+      });
+    }
+  },
+);
+
+const initialState: WalletInitialState = {
+  data: null,
+  status: undefined,
+  errorString: undefined,
+};
+
+const walletLyraSlice = createSlice({
+  name: "walletLyra",
+  initialState,
+  reducers: {
+    resetLyraAction: () => initialState,
+  },
+  extraReducers: (builder) => {
+    builder.addCase(fetchLyraStellarAddressAction.pending, (state) => ({
+      ...state,
+      data: null,
+      status: ActionStatus.PENDING,
+    }));
+
+    builder.addCase(
+      fetchLyraStellarAddressAction.fulfilled,
+      (state, action) => ({
+        ...state,
+        data: action.payload,
+        status: ActionStatus.SUCCESS,
```

Comment:
Thoughts on clearing errorStrings on pending and success events? I only started doing from Ledger, but think it's a good standard to add?

So the redux state is clean after someone successfully signs in

Reply:
Yes, that sounds good for `pending`. We might not need it for `success` as it will always be triggered after `pending`?